### ISSUE TITLE
Add `brotliCompressSync` 

### ISSUE THREAD

**Issue by Gusb3ll** [oven-sh/bun#6152]
### What is the problem this feature would solve?

Right now if you try to import the `brotliCompressSync` function from node:zlib we will get
```
Import named 'brotliCompressSync' not found in module 'node:zlib'
```

### What is the feature you are proposing to solve the problem?

To add more support for the compression algorithm aside from the current `gzip` and `deflate` that have been already implemented

**Comment by thienandangthanh** [comment]
This feature is implemented in #10722 to fix #267

**Comment by Jarred-Sumner** [comment]
This will release in Bun v1.1.8, which ships May 8th (tomorrow).

To get it early:
```sh
bun upgrade --canary
```
